ccmlib/common.py

```diff
@@ -65,6 +66,118 @@ class UnavailableSocketError(CCMError):
     pass
 
 
+class LockFile:
+    """
+    A class to create filesystem-based lockfiles that are not left
+    locked when process exits unexpectedly.
+    
+    This class uses locking mechanisms provided by kernel. Currently
+    that's fcntl.flock, due to its more intuitive semantics, but if
+    compatibility becomes a problem, it should be possible to switch to
+    fcntl.lockf or use platform-specific functions.
+
+    Lockfile stores PID of owning process and a "status" - which is an
+    arbitrary string that owning process can set. Format of the file is:
+    PID;status
+
+    This class is not thread safe.
+
+    Currently it is only used in ccm to prevent concurrent downloads,
+    so logging messages in __enter__ are specific to this use case.
+    If this class is ever needed somewhere else, this can be changed,
+    either by changing messages or the API.
+
+    Attributes
+    ----------
+    filename: str | bytes | os.PathLike
+        Path to a lockfile - used for logging
+    file: TextIO
+        File handle which will be used to take a lock.
+    locked: bool
+        Current status of the lock.
+    """
+    filename: str | bytes | os.PathLike
+    file: TextIO
+    locked: bool
```

Comment:
nit, better off adding `_` prefix to these member variables. as it seems they are not part of the public interface of `LockFile`. if they are, then we might want to provide read-only accessors or functions for accessing them.